### PR TITLE
Enhanced logging

### PR DESCRIPTION
**Pull request by OvidiuGi**

### CODE REVIEW

**Comment by live-happy-code-hard** [comment]
I've found a series of logs missing a context that would make them helpful when you really want to see what happened there. Please consider adding useful info in the logs context. Also make sure to put the variable info in the context and not appended to the log message. That practice helps you filter logs later based on the message, which would be complicated if you combine the log message from parts.

File: src/Command/ProgrammeImportFromCSVCommand.php [hunk 1]
```diff
@@ -55,24 +55,43 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         try {
             $this->import->importFromCSV($handler, $handlerMistakes, $numberImported, $numberOfLines);
         } catch (InvalidPathToFileException $e) {
-            $this->logger->info($e->getMessage(), ['path' => json_encode($e->getPathToFile())]);
+            $this->logger->info(
+                $e->getMessage(),
+                [
+                    'path' => json_encode($e->getPathToFile()),
```
Comment (live-happy-code-hard) [inline src/Command/ProgrammeImportFromCSVCommand.php:61]: Why did you json_encode the path? WHat was the purpose?

File: src/Command/ProgrammeImportFromAPICommand.php [hunk 2]
```diff
@@ -45,13 +46,19 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+                ['commandName' => self::$defaultName]
+            );
 
             return Command::FAILURE;
         }
 
         $io->success($numberImported . ' / ' . count($data) . ' programmes imported!');
-        $this->logger->info($numberImported . ' / ' . count($data) . ' programmes imported!');
+        $this->logger->info(
+            $numberImported . ' / ' . count($data) . ' programmes imported!',
```
Comment (live-happy-code-hard) [inline src/Command/ProgrammeImportFromAPICommand.php:59]: Message should be a static text and clearly state the failure.

File: src/Command/CreateBuildingCommand.php [hunk 3]
```diff
@@ -47,6 +50,14 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $this->entityManager->persist($building);
         $this->entityManager->flush();
 
+        $this->logger->info(
+            'Successfully created building: ',
```
Comment (live-happy-code-hard) [inline src/Command/CreateBuildingCommand.php:54]: The colon at the end of the message has no sense, the log will be processed and the context most probably wont be just printed after the message.

File: src/Command/CreateAccountCommand.php [hunk 4]
```diff
@@ -84,13 +87,29 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
+            );
 
             return self::FAILURE;
         }
 
         $this->entityManager->persist($user);
         $this->entityManager->flush();
 
+        $this->logger->info(
+            'Successfully created account by command: ',
```
Comment (live-happy-code-hard) [inline src/Command/CreateAccountCommand.php:106]: The colon at the end of the message is not needed

File: src/Controller/ForgotMyPasswordController.php [hunk 5]
```diff
@@ -90,17 +91,18 @@ public function reset(Request $request): Response
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            if (\date_diff(new \DateTime('now'), $user->getForgotPasswordTokenTime())->i > 60) {
+            if (date_diff(new \DateTime('now'), $user->getForgotPasswordTokenTime())->i > 60) {
```
Comment (calinpristavu) [inline src/Controller/ForgotMyPasswordController.php:94]: there are methods on the `\DateTime` class to perform diffs.

don't ever use `\date_*` functions, but instead use their OOP alternatives implemented on the `\DateTime` class

File: src/Controller/ForgotMyPasswordController.php [hunk 6]
```diff
@@ -90,17 +91,18 @@ public function reset(Request $request): Response
...
+                $this->logger->info('The reset password link expired', ['user' => $user->email]);
+
                 return new Response('The link expired', Response::HTTP_UNAUTHORIZED);
             }
 
             $password = $form->getData()['password'];
             $user->setPassword($this->passwordHasher->hashPassword($user, $password));
             $this->userRepository->add($user);
+            $this->logger->info('Password changed for ' . $user->email);
```
Comment (calinpristavu) [inline src/Controller/ForgotMyPasswordController.php:103]: why not set the user email in the context?

**Review by calinpristavu — CHANGES_REQUESTED** [review]

**Pull request merged**